pkg/server/registry/credentials/strategy.go

```diff
@@ -37,3 +37,16 @@ func (s *Strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	cred := obj.(*apiv1.Credential)
 	cred.ServerAddress = normalizeDockerIO(cred.ServerAddress)
 }
+func (s *Strategy) Validate(ctx context.Context, obj runtime.Object) (result field.ErrorList) {
+	params := obj.(*apiv1.Credential)
+	if !params.SkipValidate {
+		if err := s.credentialValidate(ctx, params.Username, *params.Password, params.ServerAddress); err != nil {
+			result = append(result, field.Invalid(field.NewPath("namespace"), params.Namespace, err.Error()))
```

Comment:
Whatever error you right here will show up to the end user. Here's what I get now if I enter bad credentials:
```
Error: Credential.api.acorn.io "quay.io" is invalid: namespace: Invalid value: "acorn": GET https://quay.io/v2/auth?scope=pull&scope=push%2Cpull&service=quay.io: UNAUTHORIZED: Invalid Username or Password; map[]
```

That `namespace: Invalid value: "acorn"` bit doesn't make sense here, nor is "Invalid" quite the right type for the error. After looking at the different functions in the `field` package, here's the best fit I could come up with:
```
result = append(result, field.Forbidden(field.NewPath("username/password"), err.Error()))
```